### PR TITLE
Cloudp 93925 add role for operator db

### PR DESCRIPTION
**Pull request by killian2k**
Add role, serviceaccount and role binding for the mongodb database
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you signed all of your commits?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: test/e2e/setup/setup.go [hunk 1]
```diff
@@ -158,6 +158,17 @@ func deployOperator(ctx *e2eutil.Context) error {
 	}
 	fmt.Println("Successfully created the operator Role Binding")
 
+	if err := buildKubernetesResourceFromYamlFile(ctx, path.Join(roleDir(), "role_database.yaml"), &rbacv1.Role{}, withNamespace(testConfig.namespace)); err != nil {
```
Comment (chatton) [inline test/e2e/setup/setup.go:161]: [nit] the error messages are referencing the operator rather than the database

**Review by chatton — COMMENTED** [review]
Looks great! 

Can we add a section to the release notes (this will be `0.7.1`)

**Review by irajdeep — APPROVED** [review]
LGTM

File: config/rbac/role_database.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      - secrets
+    verbs:
+      - get
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - patch
+      - delete
```
Comment (irajdeep) [inline config/rbac/role_database.yaml:19]: Q: I don't recall why it needs the delete permission?

Comment (irajdeep) [reply config/rbac/role_database.yaml:19]: got it, @chatton do you happen to know why we have the delete permission -- can we delete it from both the places ?

Comment (bznein) [reply config/rbac/role_database.yaml:19]: @irajdeep https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/cmd/versionhook/main.go#L197 Here's the incriminated line :)

Comment (irajdeep) [reply config/rbac/role_database.yaml:19]: got it -- thanks.

**Pull request merged**